### PR TITLE
Move derivative agg to module

### PR DESCRIPTION
**Pull request by nik9000**
This cleans up the derivative pipeline agg in a few days:
1. Moves it to the `aggregations` module ala #90283
2. Drops the ceremonial interface from the result ala #82273
3. Adds comprehensive REST layer tests for it ala #26220
4. Removes some `ESIntegTestCase tests` that duplicated those in the `AggregatorTestCase`.

### CODE REVIEW

File: modules/aggregations/src/test/java/org/elasticsearch/aggregations/bucket/histogram/InternalAutoDateHistogramTests.java [hunk 1]
```diff
@@ -65,12 +65,11 @@ protected SearchPlugin registerPlugin() {
         return new AggregationsPlugin();
     }
 
-    // TODO: the base test class should be able to get this from the search plugin? (^)
```
Comment (nik9000) [inline modules/aggregations/src/test/java/org/elasticsearch/aggregations/bucket/histogram/InternalAutoDateHistogramTests.java:68]: Here we're registering the high level REST client's version which isn't in the SearchPlugin.

Comment (nik9000) [reply modules/aggregations/src/test/java/org/elasticsearch/aggregations/bucket/histogram/InternalAutoDateHistogramTests.java:68]: I wonder if we should be removing these as we go. We don't need to parse the thing any more.

Comment (martijnvg) [reply modules/aggregations/src/test/java/org/elasticsearch/aggregations/bucket/histogram/InternalAutoDateHistogramTests.java:68]: I thought we also had these tests to test the xcontent serialization (similarly how we test the binary serialization)? We lose that if we drop these tests?

Comment (martijnvg) [reply modules/aggregations/src/test/java/org/elasticsearch/aggregations/bucket/histogram/InternalAutoDateHistogramTests.java:68]: (the downside is that we need to maintain the xcontent parsing in tests...)

Comment (nik9000) [reply modules/aggregations/src/test/java/org/elasticsearch/aggregations/bucket/histogram/InternalAutoDateHistogramTests.java:68]: Yeah. That's why I didn't nuke it. It's not obviously right. Before we had the high level REST client we never tested the round trip here - we'd just render xcontent and test in REST tests or specific xcontent renderings. Now we have randomized parsing. But it's quite heavy.

**Comment by nik9000** [comment]
run elasticsearch-ci/packaging-tests-windows-sample

**Comment by nik9000** [comment]
run elasticsearch-ci/packaging-tests-windows-sample

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by nik9000** [comment]
> [@nik9000](/nik9000) [nik9000](/nik9000) requested a review from [martijnvg](/martijnvg)

Ignore that. I didn't realize you'd already approved.

**Pull request merged**